src/test/java/lambda/part3/exercise/Exercise3.java

```diff
@@ -0,0 +1,102 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+class Exercise3 {
+
+    private static class LazyMapHelper<T, R> {
+
+        private final List<T> list;
+        private final Function<T, R> mapping;
+
+        private LazyMapHelper(List<T> list, Function<T, R> mapping){
+            this.list = list;
+            this.mapping = mapping ;
+        }
+
+        public static <T> LazyMapHelper<T, T> from(List<T> list) {
+                return
+                        new LazyMapHelper<> (list, x->x);
+        }
+
+        public List<R> force() {
+            List<R> result = new ArrayList<>();
+            for (T item : list)
+                result.add(mapping.apply(item));
+            return result;
```

Comment:
you can use `list.forEach`

Reply:
I can't do it by easy way, because foreach requires consumer but maping is function

Reply:
although..             list.forEach(item ->result.add(mapping.apply(item)));
yes, you are right)